src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +158,1547 @@ ActorInit En_Jso_InitVars = {
+            knockbackVelocity.x = 0.0f;
+            knockbackVelocity.y = 0.0f;
+            knockbackVelocity.z = -10.0f;
+            Matrix_MultVec3f(&knockbackVelocity, &this->knockbackVelocity);
+            this->swordState = EN_JSO_SWORD_STATE_KNOCKED_OUT_OF_HANDS;
+            this->disableBlure = true;
+            EnJso_SetupKnockedBack(this);
+            return;
+        }
+    }
+
+    if (curFrame >= this->animEndFrame) {
+        this->actor.speed = 0.0f;
+        this->disableBlure = true;
+        EnJso_SetupWaitAfterSlash(this);
+    }
+}
+
+void EnJso_SetupWaitAfterSlash(EnJso* this) {
+    if (this->swordHitSomething) {
+        EnJso_ChangeAnim(this, EN_JSO_ANIM_SLASH_LOOP);
+        this->timer = 20;
+    } else {
+        EnJso_ChangeAnim(this, EN_JSO_ANIM_SLASH_LOOP);
+        this->timer = 40;
+    }
+
+    this->action = EN_JSO_ACTION_WAIT_AFTER_SLASH;
+    this->actionFunc = EnJso_WaitAfterSlash;
+}
+
+/**
+ * Waits either 20 or 40 frames, depending on whether or not the Garo hit the player with its slash attack. Once this
+ * waiting period ends, the Garo starts circling the player again.
+ */
+void EnJso_WaitAfterSlash(EnJso* this, PlayState* play) {
+    if (this->timer == 0) {
+        this->attackWaitTimer = Rand_S16Offset(30, 30);
+        this->bodyCollider.base.colType = COLTYPE_NONE;
+        this->bodyCollider.base.acFlags |= AC_HARD;
+        this->swordHitSomething = false;
+        sIsAttacking = false;
+        this->isAttacking = false;
+        EnJso_SetupCirclePlayer(this, play);
+    }
+}
+
+void EnJso_SetupKnockedBack(EnJso* this) {
+    EnJso_ChangeAnim(this, EN_JSO_ANIM_KNOCKED_BACK);
+    this->bodyCollider.base.acFlags &= ~AC_HARD;
+    this->timer = 30;
+    this->action = EN_JSO_ACTION_KNOCKED_BACK;
+    this->actor.speed = 0.0f;
+    this->actionFunc = EnJso_KnockedBack;
+}
+
+/**
+ * Plays the knocked back animation to completion, then transitions to cowering on the ground.
+ */
+void EnJso_KnockedBack(EnJso* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    if (curFrame >= this->animEndFrame) {
+        EnJso_SetupCower(this);
+    }
+}
+
+void EnJso_SetupCower(EnJso* this) {
+    EnJso_ChangeAnim(this, EN_JSO_ANIM_COWER);
+    this->action = EN_JSO_ACTION_COWER;
+    this->timer = 30;
+    this->actionFunc = EnJso_Cower;
+}
+
+/**
+ * Cower on the ground for 30 frames, then jump back.
+ */
+void EnJso_Cower(EnJso* this, PlayState* play) {
+    if (this->timer == 0) {
+        EnJso_SetupJumpBack(this);
+    }
+}
+
+/**
+ * Unlike most enemies, this will make the Garo play its damaged animation rather than stopping all animations.
+ */
+void EnJso_SetupStunned(EnJso* this) {
+    Vec3f knockbackVelocity;
+
+    AudioSfx_SetChannelIO(&this->actor.projectedPos, NA_SE_EN_ANSATSUSYA_DASH_2, 0);
+    EnJso_ChangeAnim(this, EN_JSO_ANIM_DAMAGED);
+
+    //! @bug: This assignment is immediately overriden below.
+    this->timer = 30;
```

Comment:
While this is a little jank, I'm not sure this is really a bug since it doesn't affect the behaviour in any way?